src/nccl_ofi_api.c

```diff
@@ -309,15 +309,11 @@ ncclResult_t nccl_net_ofi_regMr(void *comm, void *data, size_t size, int type,
 	}
 
 	switch (base_comm->type) {
-	case NCCL_NET_OFI_SEND_COMM:;
-		nccl_net_ofi_send_comm_t *send_comm =
-			(nccl_net_ofi_send_comm_t *)base_comm;
-		ret = send_comm->regMr(send_comm, data, size, type, mhandle);
+	case NCCL_NET_OFI_SEND_COMM:
+		ret = ((nccl_net_ofi_send_comm_t *)base_comm)->regMr((nccl_net_ofi_send_comm_t *)base_comm, data, size, type, mhandle);
```

Comment:
eww...  do not want.

Reply:
totally agree, this is an artifact of our bootleg object inheritance in C, and this is a step in the direction of being able to deprecate that and do things in a simpler way. We'll fix it soon, but asking kindly that you let this suffice for now so that we can review that work separately.

Reply:
i really don't like this.

Reply:
I'll try to refactor this to look less gross. My concern is:

1. moving the declaration above the switch statement runs afoul of strict aliasing rules and the compiler is then (afaiui) free to refactor this into garbage. I don't know what to expect there, but prudence tells me it's bad news to declare two pointers to two different types that resolve to the same pointer in the same function and scope.

2. this block is already relying on the existence of a semicolon after the case statement to silence a separate warning about declaring a variable as the first statement in a switch statement as not legal. I don't know why that warning was introduced by the gcc developers, and I don't know why putting a semicolon after the switch statement is enough to make it go away, and I really don't know we hid that band-aid in plain sight in this function without as much as a comment, but doing this gross casting at least fixes that warning and lets me remove the semicolon.